src/main/java/picard/sam/CreateSequenceDictionary.java

```diff
@@ -111,62 +108,58 @@
      */
     private static final Pattern ALTERNATIVE_CONTIG_NAME_PATTERN = Pattern.compile("[0-9A-Za-z][0-9A-Za-z\\*\\+@\\|\\-]*");
-    
-    /**  
+
+    /**
      * 'AN' attribute in the dictionary
      * TODO: replace "AN" with a constant : see https://github.com/samtools/htsjdk/pull/956/files
      */
     private static final String AN_ATTRIBUTE = "AN";
 
-    public CreateSequenceDictionary() {
-        try {
-            md5 = MessageDigest.getInstance("MD5");
-        } catch (NoSuchAlgorithmException e) {
-            throw new PicardException("MD5 algorithm not found", e);
-        }
-    }
-
-    public static void main(final String[] argv) {
-        System.exit(new CreateSequenceDictionary().instanceMain(argv));
-    }
-
     /**
      * Read all the sequences from the given reference file, and convert into SAMSequenceRecords
+     *
      * @param referenceFile fasta or fasta.gz
      * @return SAMSequenceRecords containing info from the fasta, plus from cmd-line arguments.
      */
-    @Deprecated
     public SAMSequenceDictionary makeSequenceDictionary(final File referenceFile) {
-        final ReferenceSequenceFile refSeqFile =
-                ReferenceSequenceFileFactory.getReferenceSequenceFile(referenceFile, TRUNCATE_NAMES_AT_WHITESPACE);
-        ReferenceSequence refSeq;
+        final Iterable<SAMSequenceRecord> samSequenceRecordsIterable = getSamSequenceRecordsIterable();
+
         final List<SAMSequenceRecord> ret = new ArrayList<>();
         final Set<String> sequenceNames = new HashSet<>();
-        for (int numSequences = 0; numSequences < NUM_SEQUENCES && (refSeq = refSeqFile.nextSequence()) != null; ++numSequences) {
-            if (sequenceNames.contains(refSeq.getName())) {
-                throw new PicardException("Sequence name appears more than once in reference: " + refSeq.getName());
+        for (SAMSequenceRecord rec : samSequenceRecordsIterable) {
+
+            if (sequenceNames.contains(rec.getSequenceName())) {
+                throw new PicardException("Sequence name appears more than once in reference: " + rec.getSequenceName());
             }
-            sequenceNames.add(refSeq.getName());
-            ret.add(makeSequenceRecord(refSeq));
+            sequenceNames.add(rec.getSequenceName());
+            ret.add(rec);
         }
         return new SAMSequenceDictionary(ret);
     }
 
+    private Iterable<SAMSequenceRecord> getSamSequenceRecordsIterable() {
+        final SequenceDictionaryUtils.SamSequenceRecordsIterator samSequenceRecordIterator =
+                new SequenceDictionaryUtils.SamSequenceRecordsIterator(REFERENCE_SEQUENCE,
+                        TRUNCATE_NAMES_AT_WHITESPACE);
+        samSequenceRecordIterator.setGenomeAssembly(GENOME_ASSEMBLY);
+        samSequenceRecordIterator.setSpecies(SPECIES);
+        samSequenceRecordIterator.setUri(URI);
+        return () -> samSequenceRecordIterator;
```

Comment:
Is there a need to create the iterator at call time? You can defer creation until the returned iterator is called, e.g. 
```
        return () -> {
            final SequenceDictionaryUtils.SamSequenceRecordsIterator iterator =
                    new SequenceDictionaryUtils.SamSequenceRecordsIterator(REFERENCE_SEQUENCE,
                            TRUNCATE_NAMES_AT_WHITESPACE);
            iterator.setGenomeAssembly(GENOME_ASSEMBLY);
            iterator.setSpecies(SPECIES);
            iterator.setUri(URI);
            return iterator;
        };
```

Reply:
nice suggestion.